core/src/com/geekbrains/td/MonsterEmitter.java

```diff
@@ -24,10 +59,78 @@ public void update(float dt) {
         for (int i = 0; i < activeList.size(); i++) {
             activeList.get(i).update(dt);
         }
+        generateMonsters(dt);
     }
 
     public void setup(int cellX, int cellY) {
         Monster monster = getActiveElement();
         monster.activate(cellX * 80 + 40, cellY * 80 + 40);
     }
+
+    private void generateMonsters(float dt) {
+        if (monsterTimer < 0.001f) {
+            if (!waveList.isEmpty()) {
+                currentWave = waveList.pop();
+            } else {
+                currentWave = null;
+                gameScreen.levelFinish();
+            }
+        }
+        if (currentWave != null) {
+            monsterTimer += dt;
+            if (currentWave == MonsterWaveType.SINGLE){
+                isSingleMonsterStatus = true;
```

Comment:
Why do you need `isSingleMonsterStatus` if you can always get it via `currentWave == MonsterWaveType.SINGLE`?